src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,53 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.BulldozerProducerImpl;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.ExcavatorProducerImpl;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.model.TruckProducerImpl;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<? extends Machine> machines;
+        if (type == Bulldozer.class) {
+            machines = new BulldozerProducerImpl().get();
+            return new ArrayList<>(machines);
+        }
+        if (type == Truck.class) {
+            machines = new TruckProducerImpl().get();
+            return new ArrayList<>(machines);
+        }
+        if (type == Excavator.class) {
+            machines = new ExcavatorProducerImpl().get();
+            return new ArrayList<>(machines);
+        } else {
+            return new ArrayList<>();
+        }
+    }
+
+    @Override
+    public void fill(List<? super Machine> machines, Machine value) {
+        int size = machines.size();
+        if (size != 0) {
```

Comment:
useless IF. You have `i < machines.size()` condition. It will not be met in case when `size = 0`